lib/internal/Magento/Framework/Url.php

```diff
@@ -478,8 +483,8 @@ public function getBaseUrl($params = [])
         ) {
             $this->getRouteParamsResolver()->setType(UrlInterface::URL_TYPE_DIRECT_LINK);
         }
-
-        $result = $this->_getScope()->getBaseUrl($this->_getType(), $this->_isSecure());
+        $result = $customScope ? $this->getData('scope')->getBaseUrl($this->_getType(), $this->_isSecure())
```

Comment:
Please explain the logic. Why we cannot always use `$this->getData('scope')`?

Reply:
@orlangur Because this metod have usages in many methods and not always sets "_scope"  to the dataObjects, and method _getScope() checks if _scope not exist in dataObject then they give us a hardcoded admin "_scope", so basically I added a 1 more scenario where we can get _scope from data object, and this scenario will not break some tests, i think

Reply:
I think `_getScope` can be used instead of `getData('scope')` - it's basically the same plus handling the case when scope is not set.